src/primitives.jl

```diff
@@ -214,13 +214,16 @@ anynull(X::Any) = any(_isnull, X)                              # -> Bool
 anynull(X::NullableArray) = any(X.isnull)                      # -> Bool
 function anynull{T}(X::AbstractArray{T})                       # -> Bool
     u = isa(T, Union)
-    if u && !any(f -> eval(:($T.$f)) <: Nullable || Nullable <: eval(:($T.$f)), fieldnames(T))
+    if !u && !(Nullable <: T) && !(T <: Nullable)
         return false
-    elseif !u && !(Nullable <: T) && !(T <: Nullable)
-        return false
-    else
-        return any(_isnull, X)
+    if isa(T, Union)
+        if VERSION < v"0.6.0-" && !any(S -> S <: Nullable || Nullable <: S, T.types)
```

Comment:
Ah yeah, part of that is not a statically evaluable condition, so you'd have to do nested `if`s.

Instead, at the top level outside of this function
```julia
if VERSION < v"0.6.0-dev.2107"
    _uniontypes(T::Union) = T.types
else
    _uniontypes(T::Union) = Base.uniontypes(T)
```
Then use `_uniontypes(T)` in the function here.